rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2281,11 +2281,23 @@ extern "C"
 		return RMW_RET_ERROR;
 	}
 	//Get participant pointer from node
-        if(node->implementation_identifier != eprosima_fastrtps_identifier)
-        {
-            RMW_SET_ERROR_MSG("node handle not from this implementation");
-            return RMW_RET_ERROR;
-        }
+	if(node->implementation_identifier != eprosima_fastrtps_identifier)
+	{
+	    RMW_SET_ERROR_MSG("node handle not from this implementation");
+	    return RMW_RET_ERROR;
+	}
+	if (topic_names_and_types->topic_count) {
```

Comment:
I guess the patch is doing the same as the other rmw impl. atm. So it can stay like this I guess. (Still not convinced that this is a nice API...)